cmd/serverless/main.go

```diff
@@ -116,6 +117,12 @@ func runAgent(stopCh chan struct{}) (serverlessDaemon *daemon.Daemon, err error)
 
 	outputDatadogEnvVariablesForDebugging()
 
+	startChecker := startchecker.InitStartChecker()
+	startChecker.AddRule(&startchecker.ApiKeyEnvRule{})
+	if !startChecker.Check() {
+		return nil, log.Errorf("Can't start the extension: %s", startChecker.GetErrorMessage())
```

Comment:
Also, maybe I'm misunderstanding, but won't this cause the customer's lambda function to also fail? iirc we don't want to crash the function if the extension can't load. I think if at any point an extension returns/exits, the customer code also fails? (but I'm not sure).

Reply:
exact, since it's now a breaking change, I reverted that to add a no-op registration